QRCoder/QRCodeData.cs

```diff
@@ -121,60 +132,65 @@ public QRCodeData(byte[] rawData, Compression compressMode)
     /// <returns>Returns the raw data of the QR code as a byte array.</returns>
     public byte[] GetRawData(Compression compressMode)
     {
-        var bytes = new List<byte>();
+        using var output = new MemoryStream();
+        Stream targetStream = output;
+        DeflateStream? deflateStream = null;
+        GZipStream? gzipStream = null;
 
-        //Add header - signature ("QRR")
-        bytes.AddRange(new byte[] { 0x51, 0x52, 0x52, 0x00 });
-
-        //Add header - rowsize
-        bytes.Add((byte)ModuleMatrix.Count);
-
-        //Build data queue
-        var dataQueue = new Queue<int>();
-        foreach (var row in ModuleMatrix)
+        //Set up compression stream if needed
+        if (compressMode == Compression.Deflate)
         {
-            foreach (var module in row)
-            {
-                dataQueue.Enqueue((bool)module ? 1 : 0);
-            }
+            deflateStream = new DeflateStream(output, CompressionMode.Compress, true);
+            targetStream = deflateStream;
         }
-        for (int i = 0; i < 8 - (ModuleMatrix.Count * ModuleMatrix.Count) % 8; i++)
+        else if (compressMode == Compression.GZip)
         {
-            dataQueue.Enqueue(0);
+            gzipStream = new GZipStream(output, CompressionMode.Compress, true);
+            targetStream = gzipStream;
         }
 
-        //Process queue
-        while (dataQueue.Count > 0)
+        try
         {
-            byte b = 0;
-            for (int i = 7; i >= 0; i--)
+            //Add header - signature ("QRR")
+            targetStream.Write(new byte[] { 0x51, 0x52, 0x52, 0x00 }, 0, 4);
```

Comment:
Let's use collection expression to avoid the `byte[]` allocation for .NET 8+ (where they were introduced).
```suggestion
#if NET8_OR_GREATER
            targetStream.Write([0x51, 0x52, 0x52, 0x00]);
#else
            targetStream.Write(new byte[] { 0x51, 0x52, 0x52, 0x00 }, 0, 4);
#endif
```

I thinks .NET 8+ is finde, as older targets aren't supported anymore by .NET.
Else, we could keep the signature in a static readonly and load it from there. But not so nice code...

Also .NET 10's JIT would stack allocate that array, so no heap allocation.
But the collection expression reads much nicer.

Reply:
I bumped the C# version to 12, so the code you supplied will work as far back as .NET Core 2.1 -- anything supported by Span.  I'll update the conditional again to HAS_SPAN in a future PR, and probably add netstandard2_1 as a target at the same time.